core/modules/date/date.field.inc

```diff
@@ -340,6 +354,105 @@ function date_field_widget_info() {
 function date_field_widget_form(&$form, &$form_state, $field, $instance, $langcode, $items, $delta, $base) {
   module_load_include('inc', 'date', 'date.elements');
 
+  // The HTML5 widget differs quite a bit.
+  if ($instance['widget']['type'] == 'date_h5widget') {
+    if (!isset($form['#entity'])) {
+      // Prevent additional default value and useless token browser on the
+      // settings form.
+      $form['#access'] = FALSE;
+      return;
+    }
+    $field_settings = $field['settings'];
+    $widget_settings = $instance['widget']['settings'];
+
+    // We have to take some liberties here: only two types of "granularity",
+    // date with or without time. Time eventually without seconds.
```

Comment:
Curious where these limitations come from?

Reply:
I probably don't understand the question...

The date fields can have weird, odd granularity settings, but the date and time html5 inputs are strictly tied to - guess what - date and time. A date must be a full date `Y-m-d`, time must be at least `H:i`. (No hour-only input, for example.)

Quite similar to the popup widget, BTW, but that just silently ignores stuff, which leads to WTF when it comes to input.

Reply:
> ...(No hour-only input, for example.)

Right. Gotcha 👍🏼